v2/tests/log_context_property_type_wchar_t_ptr_int/log_context_property_type_wchar_t_ptr_int.c

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <wchar.h>
+
+#include "c_logging/log_context_property_type.h"
+
+#include "c_logging/log_context_property_type_wchar_t_ptr.h"
+
+#define POOR_MANS_ASSERT(cond) \
+    if (!(cond)) \
+    { \
+        (void)printf("%s:%d test failed\r\n", __FUNCTION__, __LINE__); \
+        abort(); \
+    } \
+
+/* LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string */
+
+static void wchar_t_ptr_to_string_with_NULL_property_value_fails(void)
+{
+    // arrange
+    char buffer[128];
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string(NULL, buffer, sizeof(buffer));
+
+    // assert
+    POOR_MANS_ASSERT(result < 0);
+}
+
+static void wchar_t_ptr_to_string_with_NULL_buffer_and_non_zero_buffer_length_fails(void)
+{
+    // arrange
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string(L"cucu", NULL, 1);
+
+    // assert
+    POOR_MANS_ASSERT(result < 0);
+}
+
+static void wchar_t_ptr_to_string_with_NULL_buffer_and_zero_buffer_length_returns_string_length(void)
+{
+    // arrange
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string(L"cucu", NULL, 0);
+
+    // assert
+    POOR_MANS_ASSERT(result == (int)strlen("cucu"));
+}
+
+static void wchar_t_ptr_to_string_copies_the_string(void)
+{
+    // arrange
+    char buffer[4];
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string(L"bau", buffer, sizeof(buffer));
+
+    // assert
+    POOR_MANS_ASSERT(result == 3);
+    POOR_MANS_ASSERT(strcmp(buffer, "bau") == 0);
```

Comment:
>strcmp [](http://example.com/codeflow?start=21&length=6)

let's use the wchar_t function for this.

Reply:
buffer has type char* here, so use strcmp, wcscmp can't compile